src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -269,13 +312,29 @@ def create_replica(cmd, client, store_name, name, location, resource_group_name=
                                replica_creation_parameters=replica_creation_params)
 
 
-def delete_replica(cmd, client, store_name, name, resource_group_name=None):
+def delete_replica(cmd, client, store_name, name, yes=False, resource_group_name=None):
     if resource_group_name is None:
         resource_group_name, _ = resolve_store_metadata(cmd, store_name)
 
-    return client.begin_delete(resource_group_name=resource_group_name,
-                               config_store_name=store_name,
-                               replica_name=name)
+    if not yes:
+        config_store_client = cf_configstore(cmd.cli_ctx)
+        config_store = show_configstore(
+            cmd, config_store_client, store_name, resource_group_name
+        )
+        replicas = list_replica(cmd, client, store_name, resource_group_name)
+
+        if config_store.sku.name.lower() == "premium" and len(list(replicas)) == 1:
```

Comment:
What if the provided replica name does not match the last replica name? The operation should fail without the need to display this warning.

Reply:
Currently when you want to delete a replica and fail to provide the replica name that matches the last replica name/replica name that exists. We still ask for user confirmation and the operation doesn't fail.
I was thinking we should still show the warning if we also ask for confirmation with the current implementation. What do you think

Reply:
ok, thanks! we can show the warning even if name does not match